src/features/offer/pages/Offer/Offer.tsx

```diff
@@ -38,12 +48,29 @@ export const Offer: FunctionComponent = () => {
   })
 
   const { data: offer } = useOffer({ offerId })
-  const similarOffers = useSimilarOffers(offerId, offer?.venue.coordinates)
-  const hasSimilarOffers = similarOffers && similarOffers.length > 0
+  const { data } = useSubcategories()
+  const { shouldUseAlgoliaRecommend } = useRemoteConfigContext()
+  const subcategorySearchGroupId = getSearchGroupIdFromSubcategoryId(data, offer?.subcategoryId)
+  const sameCategorySimilarOffers = useSimilarOffers({
+    offerId,
+    position: offer?.venue.coordinates,
+    shouldUseAlgoliaRecommend,
+    categoryIncluded: subcategorySearchGroupId || SearchGroupNameEnumv2.NONE,
+  })
+  const hasSameCategorySimilarOffers = sameCategorySimilarOffers?.length
```

Comment:
Du coup ici pareil :
```suggestion
  const hasSameCategorySimilarOffers = Boolean(sameCategorySimilarOffers?.length)
```